src/scripts/main.js

```diff
@@ -1 +1,62 @@
 'use strict';
+
+document.addEventListener('contextmenu', (e) => e.preventDefault());
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
```

Comment:
The first promise has a potential issue where it might not resolve if the left mouse button is not clicked within 3 seconds. This could lead to an unhandled promise rejection if the event doesn't occur in time. Consider ensuring that either the event occurs or the rejection is handled properly.